gazebo_ros2_control/src/gazebo_system.cpp

```diff
@@ -0,0 +1,485 @@
+          "Using the first entry");
+    }
+    std::string hardware_interface = joint_interfaces.front();
+    // Decide what kind of command interface this actuator/joint has
+    if (hardware_interface == "hardware_interface/EffortJointInterface") {
+      this->dataPtr->joint_control_methods_[j] = EFFORT;
+    } else if (hardware_interface == "hardware_interface/PositionJointInterface") {
+      this->dataPtr->joint_control_methods_[j] = POSITION;
+    } else if (hardware_interface == "hardware_interface/VelocityJointInterface") {
+      this->dataPtr->joint_control_methods_[j] = VELOCITY;
+    } else {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "No matching joint interface '" <<
+          hardware_interface << "' for joint " << joint_name);
+      RCLCPP_INFO(
+        this->nh_->get_logger(),
+        "    Expecting one of 'hardware_interface/{EffortJointInterface |"
+        " PositionJointInterface | VelocityJointInterface}'");
+      return false;
+    }
+
+    //
+    // Accept this URDF joint as valid and link with the gazebo joint of the same name
+    //
+
+    // I think it's safe to only skip this joint and not abort if there is no match found
+    gazebo::physics::JointPtr simjoint = parent_model->GetJoint(joint_name);
+    if (!simjoint) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Skipping joint in the URDF named '" << joint_name <<
+          "' which is not in the gazebo model.");
+      continue;
+    }
+    this->dataPtr->sim_joints_.push_back(simjoint);
+
+    // Accept this joint and continue configuration
+    RCLCPP_INFO_STREAM(
+      this->nh_->get_logger(), "Loading joint '" << joint_name << "' of type '" <<
+        hardware_interface << "'");
+
+    // register the state handles
+    this->dataPtr->joint_pos_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+      joint_name, hardware_interface::HW_IF_POSITION, &this->dataPtr->joint_position_[j]);
+    this->dataPtr->joint_vel_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+      joint_name, hardware_interface::HW_IF_VELOCITY, &this->dataPtr->joint_velocity_[j]);
+    this->dataPtr->joint_eff_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+      joint_name, hardware_interface::HW_IF_EFFORT, &this->dataPtr->joint_effort_[j]);
+
+    // register the state handles
+    this->dataPtr->joint_pos_cmd_[j] = std::make_shared<hardware_interface::CommandInterface>(
+      joint_name, hardware_interface::HW_IF_POSITION, &this->dataPtr->joint_position_cmd_[j]);
+    this->dataPtr->joint_vel_cmd_[j] = std::make_shared<hardware_interface::CommandInterface>(
+      joint_name, hardware_interface::HW_IF_VELOCITY, &this->dataPtr->joint_velocity_cmd_[j]);
+    this->dataPtr->joint_eff_cmd_[j] = std::make_shared<hardware_interface::CommandInterface>(
+      joint_name, hardware_interface::HW_IF_EFFORT, &this->dataPtr->joint_effort_cmd_[j]);
+
+    try {
```

Comment:
as the transmissions are not fully baked yet, I would advise to leave out that logic. You're declaring quite a few parameters here on that node, which are not useful at the moment. 
Without that logic, I believe you also don't need a node pointer in the first place.

Reply:
I don't get this one. Do you want me to remove the position and velocity PID ?